src/tests/GfxBatchHbaoTest.cpp

```diff
@@ -0,0 +1,413 @@
+#include <Corrade/Containers/Optional.h>
+#include <Corrade/PluginManager/Manager.h>
+#include <Corrade/Utility/Algorithms.h>
+#include <Corrade/Utility/FormatStl.h>
+#include <Corrade/Utility/Path.h>
+#include <Magnum/DebugTools/CompareImage.h>
+#include <Magnum/GL/Context.h>
+#include <Magnum/GL/Extensions.h>
+#include <Magnum/GL/Framebuffer.h>
+#include <Magnum/GL/Mesh.h>
+#include <Magnum/GL/OpenGLTester.h>
+#include <Magnum/GL/Renderer.h>
+#include <Magnum/GL/Texture.h>
+#include <Magnum/GL/TextureFormat.h>
+#include <Magnum/GL/Version.h>
+#include <Magnum/Image.h>
+#include <Magnum/ImageView.h>
+#include <Magnum/Math/Angle.h>
+#include <Magnum/Math/Color.h>
+#include <Magnum/Math/Matrix4.h>
+#include <Magnum/MeshTools/Compile.h>
+#include <Magnum/PixelFormat.h>
+#include <Magnum/Shaders/PhongGL.h>
+#include <Magnum/Trade/AbstractImporter.h>
+#include <Magnum/Trade/ImageData.h>
+#include <Magnum/Trade/MeshData.h>
+
+#include "esp/gfx_batch/Hbao.h"
+
+#include "configure.h"
+
+namespace {
+
+namespace Cr = Corrade;
+namespace Mn = Magnum;
+using namespace Mn::Math::Literals;
+
+constexpr const char* baseTestFilename = "van-gogh-room";
+
+const std::string SourceTestPlyDir =
```

Comment:
Just FYI, `Utility::Path::join()`, `Utility::format()` all return `Containers::String`, and the APIs taking strings use `StringView`s.

Not that I'd want that changed away from `std::string` across all of Habitat, but existing code in `gfx_batch` is using the Corrade ones exclusively so it'd be great if it was kept consistent. (I'm probably just annoying at this point, sorry.)

Reply:
Not at all. I wonder about these things, but I don't want to overly annoy you with questions about these little things. :)

Reply:
I'll revisit this in HBAO version 1.1 :)